src/serialization/markup.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+	Copyright (C) 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "color.hpp"
+
+// This file isn't needed by any of these functions, but this allows any
+// standard color to be passed to span_color without an extra include.
+#include "font/standard_colors.hpp"
+#include "formatter.hpp"
+#include "serialization/string_utils.hpp"
+
+#include <string>
+
+namespace markup {
+
+/**
+ * Returns the contents enclosed inside &lt;tag_name&gt; and &lt;/tag_name&gt;
+ */
+template<typename... Args>
+std::string tag(const std::string& tag_name, Args... contents)
```

Comment:
IMO it would be better for all these functions to take `Args&&...` rather than copies of their arguments.

Reply:
Also, please don't use `&lt;` and `&gt;` for tags in the documentation comments. If you need to avoid doxygen errors, wrap the tags in backticks instead. The key point is that it should be easily readable in the source code, which is definitely _not_ true of XML character entities.

Reply:
updated.